src/components/SegmentedControl/SegmentedControl.utils.ts

```diff
@@ -39,7 +39,7 @@ export function isString(element: Option | number): element is string {
  * @returns {boolean} `true` if the option is disabled, `false` otherwise.
  */
 export function isDisabledOption(option: Option, isDisabled: boolean): boolean {
-  return isDisabled ?? (!isString(option) && option?.isDisabled)
+  return (!isString(option) && option?.isDisabled) || isDisabled
```

Comment:
Why those change? It is required?
I think that the previous check is correct, if the component is disabled globally, so it should exit for isDisabled. Otherwise, it checks inside the option. 
What do you think about? Maybe it works only changing the nullish operator with the or?

Reply:
yes, we can favor isDisabled first